extensions/time-logger/src/gcalTimeLogger.tsx

```diff
@@ -0,0 +1,127 @@
+import { Action, ActionPanel, Icon, List, Detail } from "@raycast/api";
+import { useState } from "react";
+import { useEpics } from "./hooks/useEpics";
+import { useEpicInProgress } from "./hooks/useEpicInProgress";
+import { useMigrationManager } from "./hooks/useMigrationManager";
+import { useEpicFilter } from "./hooks/useEpicFilter";
+import { TimeLogAction } from "./components/TimeLogAction";
+
+export default function gcalTimeLogger() {
+  const [inputEpicName, setInputEpicName] = useState("");
+  const { epics, deleteEpic, addEpic, updateLastUsedTimestamp, updateEpic } = useEpics();
+  const { workingOnEpicData, setWorkingOnEpicData, startWork, workStartedAt } =
+    useEpicInProgress(updateLastUsedTimestamp);
+  const isMigrationNeeded = useMigrationManager();
+
+  const sortedEpics = useEpicFilter(epics, workingOnEpicData?.name, inputEpicName);
+
+  if (isMigrationNeeded === true) {
+    return <Detail isLoading={isMigrationNeeded === undefined} markdown={`# Setting up...`} />;
+  }
+
+  const createEpicFromQuery = () => {
+    if (epics) {
+      addEpic(inputEpicName);
+    }
+    setInputEpicName("");
+  };
+
+  return (
+    <>
+      <List
+        filtering={false}
+        onSearchTextChange={setInputEpicName}
+        searchText={inputEpicName}
+        searchBarPlaceholder="Epic name / description (optional)"
+        navigationTitle="Log time"
+        isLoading={!epics || workingOnEpicData === undefined}
+      >
+        {sortedEpics && sortedEpics.length > 0 ? (
+          sortedEpics.map((epic) => (
+            <List.Item
+              key={epic.name}
+              title={epic.name}
+              icon={Icon.PlayFilled}
+              subtitle={epic.description}
+              accessories={
+                workingOnEpicData?.name === epic.name && workStartedAt
+                  ? [
+                      {
+                        icon: Icon.PlayFilled,
+                        text: `Work started at ${String(workStartedAt.getHours()).padStart(2, "0")}:${String(
+                          workStartedAt.getMinutes(),
+                        ).padStart(2, "0")}`,
+                      },
+                    ]
+                  : []
+              }
+              actions={
+                <ActionPanel title="Epic">
+                  {workingOnEpicData?.name !== epic.name ? (
+                    <>
+                      <Action
+                        key="start-work"
+                        icon={Icon.PlayFilled}
+                        title="Start Working"
+                        onAction={() => startWork(epic.name)}
+                        shortcut={{ modifiers: ["cmd"], key: "g" }}
+                      />
+                      <Action
+                        key="delete-epic"
+                        icon={Icon.XMarkCircle}
+                        title="Delete This Epic"
```

Comment:
```suggestion
                        title="Delete This Epic"
                        style={Action.Style.Destructive}
```

Reply:
You can also combine it with a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert)